assets/javascripts/modules/autoComplete.js

```diff
@@ -314,28 +323,99 @@ var suggestionsEvent = function () {
   });
 };
 
+
+/**
+  * Generate random variable name to ensure no conflicts
+*/
+var randomVariableName = function () {
```

Comment:
I'd prefer not to use anything which could possibly be a conflict in the global space, so I think this should stay until the additional future work to remove the global object is completed.